js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx

```diff
@@ -453,167 +622,35 @@ const AssetGraphExplorerWithData = ({
-                          <AssetGroupNode group={group} scale={scale} />
-                        )}
-                      </foreignObject>
-                    ))}
-
-                  {Object.values(layout.nodes)
-                    .filter((node) => !isNodeOffscreen(node.bounds, viewportRect))
-                    .map(({id, bounds}) => {
-                      const graphNode = assetGraphData.nodes[id]!;
-                      const path = JSON.parse(id);
-                      if (allowGroupsOnlyZoomLevel && scale < GROUPS_ONLY_SCALE) {
-                        return;
-                      }
-                      if (bounds.width === 1) {
-                        return;
-                      }
-
-                      const contextMenuProps = {
-                        graphData: fullAssetGraphData,
-                        node: graphNode,
-                        explorerPath,
-                        onChangeExplorerPath,
-                        selectNode: selectNodeById,
-                      };
-                      return (
-                        <foreignObject
-                          {...bounds}
-                          key={id}
-                          onMouseEnter={() => setHighlighted(id)}
-                          onMouseLeave={() => setHighlighted(null)}
-                          onClick={(e) => onSelectNode(e, {path}, graphNode)}
-                          onDoubleClick={(e) => {
-                            viewportEl.current?.zoomToSVGBox(bounds, true, 1.2);
-                            e.stopPropagation();
-                          }}
-                          style={{overflow: 'visible'}}
-                        >
-                          {!graphNode ? (
-                            <AssetNodeLink assetKey={{path}} />
-                          ) : scale < MINIMAL_SCALE ? (
-                            <AssetNodeContextMenuWrapper {...contextMenuProps}>
-                              <AssetNodeMinimal
-                                definition={graphNode.definition}
-                                selected={selectedGraphNodes.includes(graphNode)}
-                                height={bounds.height}
-                              />
-                            </AssetNodeContextMenuWrapper>
-                          ) : (
-                            <AssetNodeContextMenuWrapper {...contextMenuProps}>
-                              <AssetNode
-                                definition={graphNode.definition}
-                                selected={selectedGraphNodes.includes(graphNode)}
-                              />
-                            </AssetNodeContextMenuWrapper>
-                          )}
-                        </foreignObject>
-                      );
-                    })}
-                </SVGContainer>
-              )}
-            </SVGViewport>
+              {svgViewport}
+            </ContextMenuWrapper>
+          ) : (
+            svgViewport
```

Comment:
The real change here is this conditional wrapping. We only add the ContextMenu if there is more than 1 group. Alternatively I could also make ContextMenuWrapper take some kind of `canShow` prop then I don't need to move all of the existing svgViewport logic to the new intermediate variable